scripts/benchmark/tree_hash.py

```diff
@@ -1,114 +1,216 @@
+        ('pubkey', bytes48),
+        ('withdrawal_credentials', bytes32),
+        ('effective_balance', uint64),
+        ('slashed', boolean),
+        # Epoch when validator became eligible for activation
+        ('activation_eligibility_epoch', uint64),
+        # Epoch when validator activated
+        ('activation_epoch', uint64),
+        # Epoch when validator exited
+        ('exit_epoch', uint64),
+        # Epoch when validator withdrew
+        ('withdrawable_epoch', uint64),
     ]
 
+    @classmethod
+    def create_validator(cls, pubkey, withdrawal_credentials):
+        return cls(
+            pubkey=pubkey,
+            withdrawal_credentials=withdrawal_credentials,
+            effective_balance=0,
+            slashed=False,
+            activation_eligibility_epoch=FAR_FUTURE_EPOCH,
+            activation_epoch=FAR_FUTURE_EPOCH,
+            exit_epoch=FAR_FUTURE_EPOCH,
+            withdrawable_epoch=FAR_FUTURE_EPOCH,
+        )
+
 
 class State(Serializable):
     fields = [
-        ('validator_registry', List(ValidatorRecord)),
-        ('shard_and_committee_for_slots', List(List(ShardCommittee))),
-        ('latest_crosslinks', List(CrosslinkRecord)),
+        ('validators', List(Validator, VALIDATOR_REGISTRY_LIMIT)),
+        ('balances', List(uint64, VALIDATOR_REGISTRY_LIMIT)),
+        ('randao_mixes', Vector(bytes32, EPOCHS_PER_HISTORICAL_VECTOR)),
+        ('latest_block_header', BeaconBlockHeader),
+        ('eth1_data', Eth1Data),
     ]
 
 
-validator_record = ValidatorRecord(
-    pubkey=b'\x56' * 48,
-    withdrawal_credentials=b'\x56' * 32,
-    randao_commitment=b'\x56' * 32,
-    randao_layers=123,
-    status=123,
-    latest_status_change_slot=123,
-    exit_count=123,
-    poc_commitment=b'\x56' * 32,
-    last_poc_change_slot=123,
-    second_last_poc_change_slot=123,
-)
-crosslink_record = CrosslinkRecord(slot=12847, shard_block_root=b'\x67' * 32)
-crosslink_record_stubs = [crosslink_record for i in range(1024)]
+def update_tuple_item_with_fn(tuple_data,
+                              index,
+                              fn,
+                              *args):
+    """
+    Update the ``index``th item of ``tuple_data`` to the result of calling ``fn`` on the existing
+    value.
+    """
+    list_data = list(tuple_data)
+
+    try:
+        old_value = list_data[index]
+        list_data[index] = fn(old_value, *args)
+    except IndexError:
+        raise Exception(
+            "the length of the given tuple_data is {}, the given index {} is out of index".format(
+                len(tuple_data),
+                index,
+            )
+        )
+    else:
+        return tuple(list_data)
+
+
+def update_tuple_item(tuple_data,
+                      index,
+                      new_value):
+    """
+    Update the ``index``th item of ``tuple_data`` to ``new_value``
+    """
+    return update_tuple_item_with_fn(
+        tuple_data,
+        index,
+        lambda *_: new_value
```

Comment:
shouldn't this just be
```suggestion
        lambda _: new_value
```
as we don't have any `*args`?

Reply:
p sure this was copied over from the update i had for the v0.8 spec update...

the idea was to ignore any and all positional inputs to this function (and in this case just return some value to do an unconditional swap). i ran into a case where it was helpful to ignore all positional args but it would be worthwhile to revisit our usage of this idiom to see if we still need it.

a more complete solution is something like `lambda *_, **__: new_value` but we are taking a shortcut in that we don't pass any keyword args here (and expect no one else will either)

Reply:
Yeah, it was copied from Trinity :p
I opened an issue to track it: https://github.com/ethereum/trinity/issues/911